docs/configuration.md

```diff
@@ -0,0 +1,128 @@
+# Connector Configuration Reference
+
+## Configuring Named Connectors
+
+The application can programatically configure named connectors using the java
```

Comment:
Let's be super nit-picky about our docs here and make sure we're capitalizing proper nouns (Java), putting spaces between our equals in our Java code (as if it had been formatted), and just generally giving this a very careful read-over with a copy-editor like attention.

Also can we lead with a section on why people might want to do this? For example, "Sometimes it's necessary to configure multiple credential sources, different SA impersonation paths, etc. This API allows for this advanced usage." Or something similar to frame the bigger picture.

Reply:
We should also distinguish between ConnectorConfig vs ConnectionConfig (or whatever we expose) to help our users understand how the two differ.

Reply:
Will do. Once the subject matter is tight, I'll will ask for help with a thorough proof-read.